quipucords/scanner/tasks.py

```diff
@@ -0,0 +1,117 @@
+"""Celery tasks for running scans."""
+import logging
+
+import celery
+
+from api.scanjob.model import ScanJob
+from api.scantask.model import ScanTask
+from fingerprinter.runner import FingerprintTaskRunner
+from scanner.get_scanner import get_scanner
+from scanner.job import create_details_report_for_scan_job, run_task_runner
+from scanner.runner import ScanTaskRunner
+
+logger = logging.getLogger(__name__)
+
+
+def get_task_runner_class(source_type, scan_type) -> type[ScanTaskRunner]:
+    """Get the ScanTaskRunner class type for the given source and scan types.
+
+    :returns: runner class type (not an instance of the class)
+    """
+    if scan_type == ScanTask.SCAN_TYPE_CONNECT:
+        runner_class_name = "ConnectTaskRunner"
+    elif scan_type == ScanTask.SCAN_TYPE_INSPECT:
+        runner_class_name = "InspectTaskRunner"
```

Comment:
Not a blocker for this PR, but we will might need to support SCAN_TYPE_FINGERPRINT

we have the following usecases:
- POST /api/v1/reports/ https://github.com/quipucords/quipucords/blob/590cd36ea970fb59a9c37549ce1559f85e7f2a11/quipucords/api/details_report/view.py#L96-L98
- PUT /api/v1/reports/merge/ merge https://github.com/quipucords/quipucords/blob/590cd36ea970fb59a9c37549ce1559f85e7f2a11/quipucords/api/merge_report/view.py#L54-L59
- PUT/POST /api/v1/reports/merge/jobs/ merge https://github.com/quipucords/quipucords/blob/590cd36ea970fb59a9c37549ce1559f85e7f2a11/quipucords/api/merge_report/view.py#L152-L161

From the quoted code, only the last one actually would require proper celery support as it was designed to run thread based manager. (I wish we could deprecate the other ones - first one could just upload data and push to celery for processing, w/o actually returning results)

Reply:
Yes, thank you for providing that context! You may recall I asked about this last week, but I promptly forgot to research and confirm. 😅 I believe you are right; my code likely needs some minor changes to support this use case.

Reply:
You probably already thought that, but I'll say anyway: this particular topic is probably its own jira story "celery: support 'fingerprint' type scans" or smth like that.

Reply:
I've added this to the backlog to handle later: https://issues.redhat.com/browse/DISCOVERY-340